crates/sui-storage/src/mutex_table.rs

```diff
@@ -108,7 +107,7 @@ impl<K: Hash + Eq + Send + Sync + 'static, L: Lock + 'static> LockTable<K, L> {
         let lock_table: Arc<Vec<RwLock<InnerLockTable<K, L>>>> = Arc::new(
             (0..num_shards)
                 .into_iter()
-                .map(|_| RwLock::new(HashMap::with_capacity(shard_size)))
```

Comment:
what was the original motivation to preallocate the space?
is there a safety concern related to inner resizing?
I tried to run the following [test](https://gist.github.com/phoenix-o/668d47e7c49ac89bb01cad51fa950c70) that should trigger inner resizing and it works as expected

Reply:
I'm not sure - maybe @sadhansood or @andll can comment